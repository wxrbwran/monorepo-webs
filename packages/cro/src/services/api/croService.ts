import http from '@/services/http';
import { CommonData } from 'typings/global';

export default {
  // 我的团队 查询待添加的医生列表, 多选医生作为好友
  fetchBaseRoles(data: CommonData): Promise<any> {
    return http.get('base/roles', { data });
  },

  // 我的团队 查询好友列表
  fetchDoctorFriends(data: CommonData): Promise<any> {
    return http.get('doctor/friends', { data });
  },

  // 我的团队 查询好友列表
  putTeamMembers(data: CommonData): Promise<any> {
    return http.put('doctor/team/members', { data });
  },

  // 我的团队 查询好友列表
  getTeams(data: CommonData): Promise<any> {
    return http.get(`doctor/teams?data=${JSON.stringify(data)}`);
  },




};