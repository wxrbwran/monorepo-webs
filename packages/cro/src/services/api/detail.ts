import http from '@/services/http';

export default {
  fetchMedicines(data: any): Promise<any> {
    return http.get(`medicine/name?data=${JSON.stringify(data)}`);
  },
  fetchDiagnosis(data: any): Promise<any> {
    return http.get(`disease?data=${JSON.stringify(data)}`);
  },
  // fetchUploadInfo(file: object): Promise<any> {
  //     return http.get(`clinical/form/address?fileName=${file.name}`);
  // },
  fetchUploadInfo(data?: any): Promise<any> {
    return http.get(`cro/file/address?projectSid=${data.projectSid}&fileName=${data.fileName}`);
  },
  fetchTreatment(data: any): Promise<any> {
    return http.get(`treatment?data=${JSON.stringify(data)}`);
  },
  // 获取项目详情
  getCroProjectInfo(value: string): Promise<any> {
    return http.get(`research/project/detail/${value}`);
  },
  // 添加文件
  addProjectFile(data?: any): Promise<any> {
    return http.post('research/project/file', { data });
  },
  // 文件列表
  getProjectFileList(data?: any): Promise<any> {
    return http.get(`research/project/file?data=${JSON.stringify(data)}`);
  },
  // 删除文件
  deleteProfileFile(fileId: string): Promise<any> {
    return http.delete(`research/project/file/${fileId}`);
  },
  //更新项目信息
  updateCroProject(data?: any): Promise<any> {
    return http.patch('research/project', { data });
  },
  //添加加入、排除标准
  addStandard(data?: any): Promise<any> {
    return http.post('cro/project/standard', { data });
  },
  //查看加入、排除标准
  getStandard(value: string): Promise<any> {
    return http.get(`cro/project/standard/${value}`);
  },
  //获取客观量表（检查提醒）
  getObjectiveList(data?: any): Promise<any> {
    return http.get(`research/objective/examination/scale/${data}`);
  },
  //预览文件
  getFileInfo(fileId: any): Promise<any> {
    return http.get(`research/project/file/info/${fileId}`);
  },
};
