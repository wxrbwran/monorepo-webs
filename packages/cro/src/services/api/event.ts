import http from '@/services/http';

export default {
  // 添加项目终点事件
  addEndEvent(data?: any): Promise<any> {
    return http.post('research/project/end_event',{data});
  },
  // 查看项目终点事件
  fetchEndEvent(projectSid: any): Promise < any > {
    return http.get(`research/end_event/detail/${projectSid}`);
  },
  // 修改项目终点事件
  updateEndEvent(data?: any): Promise < any > {
    return http.patch('research/project/end_event', { data });
  },
  // 查询项目中所有研究者
  fetchResearchers(projectNsId: any): Promise < any > {
    return http.get(`research/project/researchers/${projectNsId}`);
  },
  // 查询项目终点事件统计信息
  fetchEventCountInfo(data?: any): Promise < any > {
    return http.get(`research/project/event_statistics?data=${JSON.stringify(data)}`);
  },
}
