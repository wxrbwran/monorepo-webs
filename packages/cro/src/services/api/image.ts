import http from '@/services/http';

export default {
  // 获取化验单检查单统计数据接口new
  // fetchImageCountNew(data: any): Promise<any> {
  //   return http.get('image/count/last', { data });
  // },
  // 获取具体类型化验单检查单图片
  // fetchImageDetail(data: any): Promise<any> {
  //   return http.get('image/detail', { data });
  // },
  // // 获取具体类型化验单检查单图片V1版接口
  // fetchImageDetailV1(data: any): Promise<any> {

  //   return http.get('image/detail/v1', { data });
  // },
  // // 修改图片旋转角度
  // patchImageDegree(data: any): Promise<any> {
  //   return http.patch('image/detail', { data });
  // },
  // // 获取自定义指标历史中所有的图片分类
  // fetchImageTypeTabs(data: any): Promise<any> {
  //   return http.get('image/type/last', { data });
  // },
  // // 获取某一项图片分类的所有指标详情
  // fetchImageIndexHistory(data: any): Promise<any> {
  //   return http.get('image/index/history/last', { data });
  // },
  // 获取图片结构化数据指标
  // fetchImageIndex(data: any): Promise<any> {

  //   return http.get('nurse/image/index', { data });
  // },
  // // 保存图片结构化数据指标
  // putImageIndex(data: any): Promise<any> {
  //   return http.put('nurse/image/index', { data });
  // },

  // // 获取图片结构化接口
  // fetchImageIndexCustom(data: any): Promise<any> {
  //   return http.get('nurse/image/index/custom', { data });
  // },
  // // 修改图片结构化数据指标
  // putImageIndexCustom(data: any): Promise<any> {
  //   return http.put('nurse/image/index/custom', { data });
  // },

  // // 获取所有子分类，即检查部位
  // fetchImageSubtypes(): Promise<any> {
  //   return http.get('image/subtypes');
  // },
  // // 搜索图片大分类或者指标
  // fetchImageIndexInfo(data: any): Promise<any> {
  //   return http.get('image/index/info', { data });
  // },
  // // 结构化
  // // 获取图片信息及结构化数据
  // putImage(data: any): Promise<any> {
  //   return http.put('image', { data });
  // },
  fetchImageIndexes(data: any): Promise<any> {
    return http.get(`image/indexes?data=${JSON.stringify(data)}`);
    // return http.get('image/indexes', { data });
  },
  // // 保存图片信息及结构化数据
  // putImageImageIndexes(data: any): Promise<any> {
  //   return http.put('image/indexes', { data });
  // },
  // 获取检查单、其它单据
  fetchImageJcdAndOther(data: any): Promise<any> {
    // return http.get('image/jcd', { data });
    return http.get(`image/jcd?data=${JSON.stringify(data)}`);
  },
  // // 新增检查单、其它单据
  // putImageJcdAndOther(data: any): Promise<any> {
  //   return http.put('image/jcd', { data });
  // },
  // // 获取检查单问题模板
  // fetchImageTopicTemplate(data: any): Promise<any> {
  //   return http.get('image/template', { data });
  // },
  // // 新增检查单问题模板   11.18新增系统/医生检查单
  // putImageTopicTemplate(data: any): Promise<any> {
  //   return http.put('image/template', { data });
  // },
  // // 新增/修改/删除检查单问题
  // postImageTemplate(data: any): Promise<any> {
  //   return http.post('image/template', { data });
  // },
  // // 修改检查单模板名称
  // patchImageTemplateName(data: any): Promise<any> {
  //   return http.patch('image/template', { data });
  // },
  // // 获取检查单检查部位或检查方法
  // fetchImageTemplatePart(): Promise<any> {
  //   return http.get('image/template/part');
  // },
  // // 根据检查部位或检查方法获取检查名称
  // fetchImageTemplateName(data: any): Promise<any> {
  //   return http.get('image/template/name', { data });
  // },
  // 结构化
};
