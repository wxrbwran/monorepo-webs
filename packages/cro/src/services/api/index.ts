export { default as auth } from './auth';
export { default as detail } from './detail';
export { default as user } from './user';
export { default as subjective } from './subjective';
export { default as project } from './project';
export { default as patientManage } from './patientManage';
export { default as report } from './report';
export { default as query } from './query';
export { default as template } from './template';
export { default as research } from './research';
export { default as base } from './base';
export { default as event } from './event';
export { default as service } from './croService';
export { default as image } from './image';