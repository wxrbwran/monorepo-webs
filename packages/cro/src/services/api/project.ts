import http from '@/services/http';

export default {
  // 创建项目
  postCroProject(data: any): Promise<any> {
    return http.post('research/project', { data });
  },
  // 获取项目列表
  getProjectList(data: any): Promise<any> {
    // return http.get(`research/all/projects?data=${JSON.stringify(data)}`);
    // return http.get(`research/all/projects?doctorId=${1381}`);
    return http.get(`research/all/projects`);
  },
  // 获取患者统计图表
  getPatientStatistic(): Promise<any> {
    return http.get(`research/patient/statistics`);
  },
  // 获取项目统计图表
  getPStatistic(): Promise<any> {
    return http.get(`research/project/statistics`);
  },
  //删除项目
  delProject(data: any): Promise<any> {
    return http.delete(`research/project/${data}`);
  },
};
