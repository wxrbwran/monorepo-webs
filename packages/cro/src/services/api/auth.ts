import { auth, ajax } from '@/services/auth';
import qs from 'qs';

export default {
  // 登录
  token(formData: CommonData): Promise<any> {
    return auth.post('token', {
      data: qs.stringify(formData)
    });
  },
};
