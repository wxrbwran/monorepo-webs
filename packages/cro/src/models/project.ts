import { Reducer } from 'redux';
import { Effect } from 'dva';
import project from '@/services/api/project';
import detail from '@/services/api/detail';
import { IPlanInfos } from '@/utils/consts';
import { patientManage, subjective } from '@/services/api';
import { Role } from '@/utils/role';

export interface ProjectModelState {
  projectList: IProjectList[],
  objectiveScaleList: IPlanInfos[],
  reverData: object,
  objectiveGroup: IGroup[],
  formName: string,
  projDetail: any,
}
interface ProjectModelType {
  namespace: 'project';
  state: ProjectModelState;
  effects: {
    fetchProjectList: Effect
    fetchObjectiveScale: Effect
    fetchGroupList: Effect
    fetchProjectDetail: Effect
  },
  reducers: {
    saveProjectList: Reducer<ProjectModelState>;
    saveObjectiveScale: Reducer<ProjectModelState>;
    setReverData:Reducer<ProjectModelState>;
    clearReverData: Reducer<ProjectModelState>;
    setGroupList: Reducer<ProjectModelState>;
    setProjectDetail: Reducer<ProjectModelState>;
  };
}
interface IProjectList {
  id: string,
  name: string,
  detail: {
    duration: number,
    intro: string,
    avatar: string,
  }
  status: string,
  doctorId: string,
  createdAt: number,
  patientCount: number,
  avgDay: number
}
export interface IGroup {
  groupId: string,
  groupName: string,
  projectSid: string
}

export const projectState = {
  projectList: [],
  objectiveScaleList: [],
  reverData:{},
  objectiveGroup: [],
  formName: '',
  projDetail: {},
}

const ProjectModel: ProjectModelType = {
  namespace: 'project',
  state: projectState,
  effects: {
    *fetchProjectList({ payload }, { call, put }) {
      const response = yield call(project.getProjectList, payload);
      window.$storage.setItem('currentProject', response.projectInfos?.[0]?.name); // navbar显示用
      yield put({
        type: 'saveProjectList',
        payload: response.projectInfos,
      });
    },
    *fetchObjectiveScale({ payload }, { call, put }) {
      const response = yield call(subjective.getObjectiveScale, payload);
      yield put({
        type: 'saveObjectiveScale',
        payload: response,
      });
    },
    *fetchGroupList({ payload }, { call, put }) {
      const response = yield call(patientManage.getGroupList, payload);
      yield put({
        type: 'setGroupList',
        payload: response.infos,
      });
    },
    *fetchProjectDetail({ payload }, { call, put }) {
      const response = yield call(detail.getCroProjectInfo, payload);
      const isLeader = [Role.MAIN_PI.id, Role.PROJECT_LEADER.id].includes(response.roleType);
      window.$storage.setItem('croLabel', response.label); // "multi_project"多中心
      window.$storage.setItem('croRoleType', response.roleType); // 科研角色
      window.$storage.setItem('projectStatus', response.status); // 项目状态
      if (isLeader) {
        window.$storage.setItem('isLeader', 'leader'); // 有些按钮功能只有项目总pi和项目组长可以操作
      } else {
        window.$storage.removeItem('isLeader');
      }
      yield put({
        type: 'setProjectDetail',
        payload: response,
      });
    },
  },
  reducers: {
    saveProjectList(state = projectState, { payload }): ProjectModelState{
      return {
        ...state,
        projectList: payload,
      };
    },
    saveObjectiveScale(state = projectState, { payload }): ProjectModelState{
      console.log('payload888',payload);
      return {
        ...state,
        objectiveScaleList: payload.infos,
        formName: payload.name
      };
    },
    setReverData(state = projectState, { payload }): ProjectModelState{
      return {
        ...state,
        reverData: Object.assign({}, state.reverData, {
          [payload.index]: {base: payload.base, plans: payload.plans},
        }),
      };
    },
    clearReverData(state = projectState, { payload }): ProjectModelState{
      return {
        ...state,
        reverData: {
          [payload.index]: {}
        },
      };
    },
    setGroupList(state = projectState, { payload }): ProjectModelState{
      return {
        ...state,
        objectiveGroup: payload,
      };
    },
    setProjectDetail(state = projectState, { payload }): ProjectModelState{
      return {
        ...state,
        projDetail: payload,
      };
    },
  },
};

export default ProjectModel;
