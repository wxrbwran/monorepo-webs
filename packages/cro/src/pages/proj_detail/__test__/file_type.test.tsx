import * as React from 'react'
import { render, RenderResult, fireEvent, waitFor } from '@testing-library/react';
import FileType from '../upload_file/file_type';
import project from '@/assets/project.png';
import * as api from '@/services/api';

let wrapper: RenderResult;
const props = {
  type: 'PROJECT_FILE',
  name: '项目文件',
  imgSrc: project,
};

jest.mock("@/services/api", () => {
  return {
    detail: {
      getProjectFileList: jest.fn(() => {
        return Promise.resolve({
          infos: [
            {
              address: "https://clinic-trial-attachments.oss-cn-beijing.aliyuncs.com/2924d525-4ca9-4c07-872d-92ca1cdcb476",
              fileId: "5B6jV5",
              name: "form1.xlsx",
              type: "PROJECT_FILE"
            }
          ], status: 'success'
        });
      }),
    }
  };
});
describe('项目详情：上传文件', () => {
  beforeEach(() => {
    wrapper = render(<FileType {...props} />)
  })
  it('匹配快照', () => {
    expect(wrapper).toMatchSnapshot();
  })
  it('点击文件夹，显示文件列表', async() => {
    const fileContent = wrapper.queryByTestId('uploadBtn');
    expect(fileContent).toBeNull();
    const toogleUpload = wrapper.getByTestId('toogleUpload');
    fireEvent.click(toogleUpload);
    await waitFor(() => {
      const uploadBtn = wrapper.getByTestId('uploadBtn');
      expect(uploadBtn).toBeInTheDocument();
    })
    // wrapper.debug();
  })
  // 显示弹框后，获取文件列表----失败
  // it('获取已上传的文件列表', async() => {
  //   const toogleUpload = wrapper.getByTestId('toogleUpload');
  //   fireEvent.click(toogleUpload);
  //   await waitFor(() => {
  //     // expect(api.detail.getProjectFileList).toHaveBeenCalledTimes(1)
  //     wrapper.debug();
  //   })
  // })
})