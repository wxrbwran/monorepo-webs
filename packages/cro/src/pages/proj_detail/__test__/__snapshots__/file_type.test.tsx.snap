// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`项目详情：上传文件 匹配快照 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="toogleUpload"
      >
        <div>
          <span
            class=""
          >
            <div
              class="ant-upload ant-upload-select ant-upload-select-text"
            >
              <span
                class="ant-upload"
                role="button"
                tabindex="0"
              >
                <input
                  accept=""
                  style="display: none;"
                  type="file"
                />
                <span
                  aria-label="upload"
                  class="anticon anticon-upload"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="upload"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                    />
                  </svg>
                </span>
                <span>
                  上传
                </span>
              </span>
            </div>
          </span>
        </div>
        <p>
          <img
            alt=""
            src="project.png"
          />
        </p>
        <p>
          项目文件
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="toogleUpload"
    >
      <div>
        <span
          class=""
        >
          <div
            class="ant-upload ant-upload-select ant-upload-select-text"
          >
            <span
              class="ant-upload"
              role="button"
              tabindex="0"
            >
              <input
                accept=""
                style="display: none;"
                type="file"
              />
              <span
                aria-label="upload"
                class="anticon anticon-upload"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="upload"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                  />
                </svg>
              </span>
              <span>
                上传
              </span>
            </span>
          </div>
        </span>
      </div>
      <p>
        <img
          alt=""
          src="project.png"
        />
      </p>
      <p>
        项目文件
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
